presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java

```diff
@@ -1062,6 +1064,52 @@ public void testOutputNotRequired()
         }
     }
 
+    @Test
+    public void testAdaptiveBatchSizes()
+            throws Exception
+    {
+        Type type = VARCHAR;
+        List<Type> types = ImmutableList.of(type);
+
+        TempFile tempFile = new TempFile();
+        List<String> values = new ArrayList<>();
+        int rowCount = 10000;
+        int longStringLength = 5000;
+        Random random = new Random();
+        for (int i = 0; i < rowCount; ++i) {
+            if (i < MAX_BATCH_SIZE) {
+                StringBuilder builder = new StringBuilder();
+                for (int j = 0; j < longStringLength; ++j) {
```

Comment:
This loop runs 5M times, right? How long does the test take to run? Perpaps, allocate byte[] array of size 5000 and write to it directly.

Reply:
The test takes 500-700ms on my mac. Generating the values takes around 100ms, which I feel it's OK. Allocating a byte[] array doesn't guarantee a fixed length of a string (can generate null in the middle).

Reply:
I was suggesting to use byte[] array to avoid incrementally building the string using StringBuilder, e.g. one could allocate byte[5000], fill it in, turn it into a string. The test runtimes though are fine, hence, no point to bother.